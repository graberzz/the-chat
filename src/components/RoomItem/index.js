import React from 'react';
import { NavLink } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import { vars } from 'components/common';

const activeClassName = 'room-item--active';

const RoomItemWrap = styled(NavLink)`
  display: block;
  padding: 10px 15px;
  transition: all .3s;

  &:not(:last-child) {
    border-bottom: 1px solid #ddd;
  }

  &.${activeClassName} {
    color: #fff;
    background-color: ${vars.mainColor};
  }
`;

const Creator = styled.div`
  color: #aaa;
  font-size: .8em;
  float: right;
`;

const RoomItem = ({ room }) => (
    <RoomItemWrap
      to={`/${room.id}`}
      activeClassName={activeClassName}
    >
      {room.name}
      <Creator>by {room.creator || 'Cthulhu'}</Creator>
    </RoomItemWrap>
);

RoomItem.propTypes = {
  room: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    creator: PropTypes.string.isRequired,
  }).isRequired,
};

export default RoomItem;