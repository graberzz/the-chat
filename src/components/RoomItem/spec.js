import React from 'react';
import { shallow } from 'enzyme';
import RoomItem from './';

describe('<RoomItem />', () => {
  it('matches snapshot', () => {
    const room = {
      id: 'id',
      name: 'test room',
      creator: 'tester',
    };

    expect(shallow(<RoomItem room={room} />)).toMatchSnapshot();
  });
});