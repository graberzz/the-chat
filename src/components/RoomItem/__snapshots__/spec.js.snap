// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomItem /> matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RoomItem
    room={
        Object {
            "creator": "tester",
            "id": "id",
            "name": "test room",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeClassName": "room-item--active",
      "children": Array [
        "test room",
        <styled.div>
          by 
          tester
</styled.div>,
      ],
      "to": "/id",
    },
    "ref": null,
    "rendered": Array [
      "test room",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            "by ",
            "tester",
          ],
        },
        "ref": null,
        "rendered": Array [
          "by ",
          "tester",
        ],
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeClassName": "room-item--active",
        "children": Array [
          "test room",
          <styled.div>
            by 
            tester
</styled.div>,
        ],
        "to": "/id",
      },
      "ref": null,
      "rendered": Array [
        "test room",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "by ",
              "tester",
            ],
          },
          "ref": null,
          "rendered": Array [
            "by ",
            "tester",
          ],
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
